### PR TITLE
[glfw] Added extrusions (E) and route (R) shortcuts

### PR DESCRIPTION
**Pull request by brunoabinader**
Useful for testing extrusions and navigational animations.

Suggested use:
1. `make run-glfw-app`
2. Press `S` to change style - works nice with night traffic
3. Press `E` to add the extrusion layer
4. Press `R` to start the route animation
5. Press any key other than `R` to stop the route animation

glfw

### CODE REVIEW

**Comment by brunoabinader** [comment]
Our Travis clang-tidy check caught a lint on cheap-ruler-cpp:
> ../../../mason_packages/headers/cheap-ruler/2.5.0/include/mapbox/cheap_ruler.hpp:202:35: error: constexpr variable 'getPoint' must be initialized by a constant expression [clang-diagnostic-error]
        constexpr auto getPoint = [](auto tuple) { return std::get<0>(tuple); };
                                  ^

As per talk with @tmpsantos - I'll fix this in cheap-ruler-cpp, bump a new version and update this PR after.

File: cmake/core-files.cmake [hunk 1]
```diff
@@ -531,6 +531,7 @@ set(MBGL_CORE_FILES
     include/mbgl/util/immutable.hpp
     include/mbgl/util/indexed_tuple.hpp
     include/mbgl/util/interpolate.hpp
+    include/mbgl/util/io.hpp
```
Comment (kkaefer) [inline cmake/core-files.cmake]: Could we please retain that as private? We could use some other way of reading the file in GLFW.

Comment (brunoabinader) [inline cmake/core-files.cmake]: Sure - my rationale when doing this was because our `read_file` implementation relies solely on C++ standard library.

**Comment by brunoabinader** [comment]
> Adding extrusions does not work for satellite layer

Right - disabling for that 👍 

> Can we make E toggle the building layer?

👍 

> Can we automatically re-add the building layer when switching styles?

👍 

> Adding the route line fails on macOS with libc++abi.dylib: terminating with uncaught exception of type std::runtime_error: Cannot read file ../../../platform/glfw/resources/new-york-route.geojson

I only tested this on Linux - my bad 👍

File: platform/glfw/glfw_view.cpp [hunk 2]
```diff
@@ -531,6 +575,52 @@ void GLFWView::setWindowTitle(const std::string& title) {
     glfwSetWindowTitle(window, (std::string { "Mapbox GL: " } + title).c_str());
 }
 
+void GLFWView::onDidFinishLoadingStyle() {
+    printf("wololo\n");
```
Comment (kkaefer) [inline platform/glfw/glfw_view.cpp]: wololo?

File: platform/glfw/ny_route.hpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
+#include <string>
+
+namespace mbgl {
+namespace platform {
+namespace glfw {
+
+constexpr const char* route = "{\
```
Comment (kkaefer) [inline platform/glfw/ny_route.hpp]: Can we please use a [raw string literal](http://en.cppreference.com/w/cpp/language/string_literal) like we do in the [shader code](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/shaders/fill.cpp#L9)?

File: platform/glfw/glfw_view.cpp (hunk 2 above)
Comment (brunoabinader) [inline platform/glfw/glfw_view.cpp]: That's my usual debug message 😂 Sorry that leaked out, removing

**Pull request merged**